gnomad/utils/vcf.py

```diff
@@ -92,3 +339,488 @@ def get_pipe_expr(array_expr: hl.expr.ArrayExpression) -> hl.expr.StringExpressi
+    :return: Dictionary keyed by VCF FILTER annotations, where values are Dictionaries of Number and Description attributes.
+    """
+    filter_dict = {
+        "AC0": {
+            "Description": "Allele count is zero after filtering out low-confidence genotypes (GQ < 20; DP < 10; and AB < 0.2 for het calls)"
+        },
+        "InbreedingCoeff": {"Description": f"InbreedingCoeff < {inbreeding_cutoff}"},
+        "RF": {
+            "Description": f"Failed random forest filtering thresholds of {snp_cutoff}, {indel_cutoff} (probabilities of being a true positive variant) for SNPs, indels"
+        },
+        "PASS": {"Description": "Passed all variant filters"},
+    }
+    return filter_dict
+
+
+def make_hist_bin_edges_expr(ht: hl.Table, prefix: str) -> Dict[str, str]:
+    """
+    Create dictionaries containing variant histogram annotations and their associated bin edges, formatted into a string
+    separated by pipe delimiters.
+
+    :param ht: Table containing histogram variant annotations.
+    :param prefix: Prefix text for age histogram bin edges. 
+    :return: Dictionary keyed by histogram annotation name, with corresponding reformatted bin edges for values.
+    """
+    edges_dict = {
+        f"{prefix}het": "|".join(
+            map(lambda x: f"{x:.1f}", ht.head(1).age_hist_het.collect()[0].bin_edges)
+        ),
+        f"{prefix}hom": "|".join(
+            map(lambda x: f"{x:.1f}", ht.head(1).age_hist_hom.collect()[0].bin_edges)
+        ),
+    }
+    for hist in HISTS:
+
+        # Parse hists calculated on both raw and adj-filtered data
+        for hist_type in ["adj_qual_hists", "qual_hists"]:
+            hist_name = hist
+            if "adj" in hist_type:
+                hist_name = f"{hist}_adj"
+            edges_dict[hist_name] = (
+                "|".join(
+                    map(
+                        lambda x: f"{x:.2f}",
+                        ht.head(1)[hist_type][hist].collect()[0].bin_edges,
+                    )
+                )
+                if "ab" in hist
+                else "|".join(
+                    map(
+                        lambda x: str(int(x)),
+                        ht.head(1)[hist_type][hist].collect()[0].bin_edges,
+                    )
+                )
+            )
+    return edges_dict
+
+
+def make_hist_dict(bin_edges: Dict[str, Dict[str, str]], adj: bool) -> Dict[str, str]:
+    """
+    Generate dictionary of Number and Description attributes to be used in the VCF header, specifically for histogram annotations.
+
+    :param bin_edges: Dictionary keyed by histogram annotation name, with corresponding string-reformatted bin edges for values.
+    :param adj: Whether to create a header dict for raw or adj quality histograms.
+    :return: Dictionary keyed by VCF INFO annotations, where values are Dictionaries of Number and Description attributes.
+    """
+    header_hist_dict = {}
+    for hist in HISTS:
```

Comment:
Just thought about this, but for all functions that use the constant HISTS, make hists a parameter with HISTS as default